pkg/services/alerting/conditions/query_test.go

```diff
@@ -242,7 +242,9 @@ func (rh fakeReqHandler) HandleRequest(context.Context, *models.DataSource, plug
 func queryConditionScenario(desc string, fn queryConditionScenarioFunc) {
 	Convey(desc, func() {
 		bus.AddHandler("test", func(query *models.GetDataSourceQuery) error {
-			query.Result = &models.DataSource{Id: 1, Type: "graphite"}
+			jsonData, err := simplejson.NewJson([]byte(`{}`))
```

Comment:
switching to `interval` fixed this automatically, because that place checks for datasource-is-nil ( https://github.com/grafana/grafana/blob/main/pkg/tsdb/interval/interval.go#L104 ) .